classes/Kohana/Log/Writer.php

```diff
@@ -11,38 +11,33 @@
 abstract class Kohana_Log_Writer {
 
 	/**
-	 * @var  string  timestamp format for log entries.
+	 * @var  string  date/time format for writing the timestamp of log entries.
 	 *
 	 * Defaults to Date::$timestamp_format
 	 */
-	public static $timestamp;
+	private $timestamp_format;
 
 	/**
-	 * @var  string  timezone for log entries
+	 * @var  string  timezone for this log writer
 	 *
 	 * Defaults to Date::$timezone, which defaults to date_default_timezone_get()
 	 */
-	public static $timezone;
+	private $timezone;
 
 	/**
-	 * Numeric log level to string lookup table.
-	 * @var array
+	 * @var  string  Level to use for stack traces
 	 */
-	protected $_log_levels = array(
-		LOG_EMERG   => 'EMERGENCY',
-		LOG_ALERT   => 'ALERT',
-		LOG_CRIT    => 'CRITICAL',
-		LOG_ERR     => 'ERROR',
-		LOG_WARNING => 'WARNING',
-		LOG_NOTICE  => 'NOTICE',
-		LOG_INFO    => 'INFO',
-		LOG_DEBUG   => 'DEBUG',
-	);
+	private $strace_level = \Psr\Log\LogLevel::DEBUG;
 
 	/**
-	 * @var  int  Level to use for stack traces
+	 * @var  string  default string format of the log entry
 	 */
-	public static $strace_level = LOG_DEBUG;
+	private $format = "time --- level: body in file:line";
+
+	/**
+	 * @var array log levels that this writer accepts to write
+	 */
+	private $levels;
```

Comment:
I'm not sure the meaning of `levels` is obvious - same for the various setters/getters. `write_levels`, `include_levels`?